### PR TITLE
[stable/sumologic-fluentd] Add support for user defined fluentd configuration

### PR DESCRIPTION
**Pull request by darend**
## what

Introduce a new variable that allows user-defined fluentd configuration files to be installed

Keys under `sumologic.fluentdUserConfig` are treated as json files to be mounted. This allows installations to extend the fluentd configuration with additional source and filters

## who

@frankreno

### CODE REVIEW

**Comment by unguiculus** [comment]
The named templates in this chart are not namespaced yet. Would you mind updating them as well for consistency?

**Comment by darend** [comment]
> The named templates in this chart are not namespaced yet. Would you mind updating them as well for consistency?

@unguiculus Happy to but I do not understand what needs to be done. Can you provide a reference or example?

**Comment by unguiculus** [comment]
You added `{{- define "sumologic.fluentdUserConfig.fullname" -}}` with the `sumologic` namespace/prefix. The other named templates in `_helpers.tpl` don't have the prefix yet but should also get it. And while you're at it: The prefix should match the name of the chart, so please change it to `sumologic-fluentd`.

**Comment by darend** [comment]
> You added {{- define "sumologic.fluentdUserConfig.fullname" -}} with the sumologic namespace/prefix. The other named templates in _helpers.tpl don't have the prefix yet but should also get it. And while you're at it: The prefix should match the name of the chart, so please change it to sumologic-fluentd.

@unguiculus https://github.com/kubernetes/charts/pull/3991/commits/76ed64a4b651aeb82c4302eb55bc84a77c0783df

File: stable/sumologic-fluentd/templates/_helpers.tpl [hunk 1]
```diff
@@ -2,22 +2,30 @@
...
-{{- printf "%s-secrets-%d" (include "fullname" .) .Release.Time.Seconds -}}
+{{- define "sumologic-fluentd.fullname-secrets" -}}
+{{- printf "%s-secrets-%d" (include "sumologic-fluentd.fullname" .) .Release.Time.Seconds -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified fluentd user conf name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "sumologic-fluentd.fluentdUserConfig.fullname" -}}
```
Comment (unguiculus) [inline stable/sumologic-fluentd/templates/_helpers.tpl:29]: This should be based on the fullname template and just add a suffix. But why do you want a different name at all? I'd just use the standard fullnsme template.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/sumologic-fluentd/templates/_helpers.tpl (hunk 1 above)
Comment (darend) [reply stable/sumologic-fluentd/templates/_helpers.tpl:29]: > This should be based on the fullname template and just add a suffix

Ahh yes, done https://github.com/kubernetes/charts/pull/3991/commits/2d094963d114c7587f02ed0ee8ae6da7a1a6f622

> But why do you want a different name at all? I'd just use the standard fullnsme template.

I was thinking they might want flexibility to support other configmaps. This one is specific to injecting the fluentd user config

**Review by unguiculus — APPROVED** [review]

**Pull request merged**